dotnet/samples/04.e.twentyQuestions/GameBot.cs

```diff
@@ -0,0 +1,91 @@
+                await turnContext.SendActivityAsync(ResponseBuilder.StartGame(), cancellationToken: cancellationToken);
+            }
+            else
+            {
+                int guessCount = turnState.Conversation!.GuessCount;
+                int remainingGuesses = turnState.Conversation!.RemainingGuesses;
+                guessCount++;
+                remainingGuesses--;
+
+                // Check correct guess
+                if (!string.IsNullOrEmpty(input) && input.Contains(secretWord, StringComparison.OrdinalIgnoreCase))
+                {
+                    await turnContext.SendActivityAsync(ResponseBuilder.YouWin(secretWord), cancellationToken: cancellationToken);
+                    secretWord = string.Empty;
+                    guessCount = 0;
+                    remainingGuesses = 0;
+                }
+                else if (remainingGuesses == 0)
+                {
+                    await turnContext.SendActivityAsync(ResponseBuilder.YouLose(secretWord), cancellationToken: cancellationToken);
+                    secretWord = string.Empty;
+                    guessCount = 0;
+                    remainingGuesses = 0;
+                }
+                else
+                {
+                    // Ask for hint
+                    string hint = await GetHint(turnContext, turnState, cancellationToken);
+                    if (hint.Contains(secretWord, StringComparison.OrdinalIgnoreCase))
+                    {
+                        await turnContext.SendActivityAsync($"[{guessCount}] {ResponseBuilder.BlockSecretWord()}", cancellationToken: cancellationToken);
+                    }
+                    else if (remainingGuesses == 1)
+                    {
+                        await turnContext.SendActivityAsync($"[{guessCount}] {ResponseBuilder.LastGuess(hint)}", cancellationToken: cancellationToken);
+                    }
+                    else
+                    {
+                        await turnContext.SendActivityAsync($"[{guessCount}] {hint}", cancellationToken: cancellationToken);
+                    }
+                }
+
+                // Save game state
+                turnState.Conversation!.SecretWord = secretWord;
+                turnState.Conversation!.GuessCount = guessCount;
+                turnState.Conversation!.RemainingGuesses = remainingGuesses;
+            }
+        }
+
+        private async Task<string> GetHint(ITurnContext turnContext, GameState turnState, CancellationToken cancellationToken)
+        {
+            // Set prompt variables
+            AI.Prompts.Variables.Add("guessCount", turnState.Conversation!.GuessCount.ToString());
+            AI.Prompts.Variables.Add("remainingGuesses", turnState.Conversation!.RemainingGuesses.ToString());
+            AI.Prompts.Variables.Add("secretWord", turnState.Conversation!.SecretWord!);
+            AI.Prompts.Variables.Add("input", turnContext.Activity.Text);
```

Comment:
This should be removed. The prompt manager's rendering will add this as a variable behind the scenes. I have pushed a fix: https://github.com/microsoft/teams-ai/pull/286

Reply:
Removed